From d1cf721a904aa298f6055c62095795601e116b06 Mon Sep 17 00:00:00 2001
From: Jared McArthur <j-mcarthur@ti.com>
Date: Thu, 10 Jul 2025 20:22:40 -0500
Subject: [PATCH 2/5] PENDING: arm64: dts: ti: k3-am62p-j722s-common-main: Fix
 interrupts property for csi2rx

Commit e73600d823c8 ("PENDING: arm64: dts: ti:
k3-am62p-j722s-common-main: Add interrupts property") added the
interrupts property to the cdns_csi2rx0 node, but added the wrong
interrupt IDs. Fix this by providing the correct interrupt IDs.

Interrupt IDs are taken from the J722S TRM [0].

Interrupt Line    | Source Interrupt
------------------|-------------------------
GICSS0_SPI_IN_175 | CSI_RX_IF0_CSI_ERR_IRQ_0
GICSS0_SPI_IN_173 | CSI_RX_IF0_CSI_IRQ_0

[0] https://www.ti.com/lit/zip/sprujb3

Fixes: e73600d823c8 ("PENDING: arm64: dts: ti: k3-am62p-j722s-common-main: Add interrupts property")
Signed-off-by: Jared McArthur <j-mcarthur@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
index 2f5ef6ff9786..e0d8b2a10b5f 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
@@ -1084,8 +1084,8 @@ ti_csi2rx0: ticsi2rx@30102000 {
 		cdns_csi2rx0: csi-bridge@30101000 {
 			compatible = "ti,j721e-csi2rx", "cdns,csi2rx";
 			reg = <0x00 0x30101000 0x00 0x1000>;
-			interrupts = <GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 156 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "error_irq", "irq";
 			clocks = <&k3_clks 182 0>, <&k3_clks 182 3>, <&k3_clks 182 0>,
 				<&k3_clks 182 0>, <&k3_clks 182 4>, <&k3_clks 182 4>;
-- 
2.50.1

